src/edge/views.py

```diff
@@ -565,21 +593,37 @@ def parse_arguments(self, request):
         parser.add_argument('design_primers', field_type=bool, required=False,
                             default=False, location='json')
         parser.add_argument('primer3_opts', field_type=dict, required=False,
-                            default={}, location='json')
+                            default=None, location='json')
+        parser.add_argument('annotations', field_type=list, required=False,
+                            default=None, location='json')
 
         args = parser.parse_args(request)
-        cassette = args['cassette']
+        cassette = args['cassette'].strip()
         homology_arm_length = args['homology_arm_length']
         genome_name = args['genome_name']
         cassette_name = args['cassette_name']
         notes = args['notes']
         design_primers = args['design_primers']
         primer3_opts = args['primer3_opts']
+        annotations = args['annotations']
+
+        if primer3_opts is None:
+            primer3_opts = {}
+        if annotations is None:
+            annotations = []
+        elif GenomeRecombinationView.validate_annotations(cassette, annotations) is False:
+            errors.append(
+                "Annotations failed validation: \
+please make sure donor sequence does not have overhangs \
+and annotation array elements have all the required fields"
```

Comment:
Can we add a period at the end of the error message since it's a sentence.

Reply:
Done